ospf6d/ospf6_asbr.c

```diff
@@ -1936,13 +1936,12 @@ DEFPY (ospf6_default_route_originate,
 	if (mtype_str == NULL)
 		mtype = -1;
 
-	/* To check ,if user is providing same route map */
-	if ((rtmap == ROUTEMAP_NAME(red))
-	    || (rtmap && ROUTEMAP_NAME(red)
-		&& (strcmp(rtmap, ROUTEMAP_NAME(red)) == 0)))
+	/* To check if user is providing same route map */
+	if (rtmap && ROUTEMAP_NAME(red)
```

Comment:
Removing this check "rtmap == ROUTEMAP_NAME(red)" will lead to below issue:
if user execute "default-information originate" command multiple times without the route-map option, the default LSAs will be re-originated. You can try it.

Reply:
You're right, but this change still looks really strange. How about changing it to `(!rtmap && !ROUTEMAP_NAME(red))`?

Reply:
Yeah changing it to the above would make more sense.

Reply:
Got! Thanks!